lenskit/batch/_multi.py

```diff
@@ -291,9 +291,12 @@ def _predict(self, rid, algo, test):
         return preds, watch.elapsed()
 
     def _recommend(self, rid, algo, test, candidates):
-        if self.recommend is None:
+        if self.recommend is None or self.recommend is False or self.recommend is 0:
             return None, None
 
+        if self.recommend is True:
+            self.recommend = None
```

Comment:
This is going to change the recommend list size for the next run-through, not what we want. We can instead do a local variable:

```python
if not self.recommend:
    return stuff
elif self.recommend is True:  # special value True means unlimited
    nrecs = None
else:  # recommend has rec size
    nrecs = self.recommend
```